### ISSUE TITLE
Tap Touch Action no longer works with Appium 1.8.0

### ISSUE THREAD

**Issue by jkorany** [appium/python-client#213]
Sending a tap event using Touch Actions no longer works with Appium version 1.8.0 and Python Client 0.26 running on a physical iOS device. [After asking on a related Appium server issue](https://github.com/appium/appium/issues/10576), it was mentioned that the Python client should not be sending `null` for x and y coordinates if the coordinates are not provided. It's suggested that the Python Client not send a x and y coordinate parameter if they aren't specified, rather than defaulting to `None`.

**Comment by trackvia-ios** [comment]
I'm running `1.7.2`, and also experiencing quite a bit of issues with touch actions.

I did have luck adding this to my capabilities for pre-existing visibility expectations to continue to work:

`"simpleIsVisibleCheck": true`

But all-in-all it sounds like we're stuck waiting for all related parties to release updates that all address the newest iOS changes for UI automation frameworks, et al.

**Comment by WaFeIS7** [comment]
Hi Everyone, any updates when this issue will be fixed? 
I have this issue with Appium 1.8.1, iOS 11.4, Appium-Python-Client (0.26).

**Comment by KazuCocoa** [comment]
What about using the latest master branch? (not 0.26)
@WaFeIS7

**Comment by WaFeIS7** [comment]
@KazuCocoa  
I will try it, a soon as possible. 
Thanks

**Comment by Madh93** [comment]
A little trick is set `x` and `y` to `0` (by default they are None).

For instance:

```python
from appium.webdriver.common.touch_action import TouchAction
# ...
actions = TouchAction(driver)
actions.tap(element, 0, 0)
actions.perform()
```

Remember: `x` and `y` coordinates **are relative** to the top left corner of the element.

**Comment by ashwinibaisutrave** [comment]
I am also facing the same issue tap on element is not working .any workaround please

**Comment by KazuCocoa** [comment]
@ashwinibaisutrave 
What version did you use? The latest one is 0.28 https://github.com/appium/python-client#appium-python-client

**Comment by ashwinibaisutrave** [comment]
I am using Java client 6.1.0.

**Comment by KazuCocoa** [comment]
I'm confusing a bit.
This issue was issuing the wrong command by python-clisent. It was fixed in #214. Python 0.28 includes the fix.

@ashwinibaisutrave 
I'm not sure Java client's implementation. Did you refer https://github.com/appium/java-client/search?q=TouchAction&type=Issues or another places like discuss.appium.io ?
If you can't find the solution, it's better to create an issue in Java client repository.


---

Anyway, this issue was fixed in 0.28. So I close this.
